app/consumer-democracy/app.go

```diff
@@ -633,6 +639,29 @@ func New(
 				fromVM[moduleName] = eachModule.ConsensusVersion()
 			}
 
+			// For a new consumer chain, this code (together with the entire SetUpgradeHandler) is not needed at all,
+			// upgrade handler code is application specific. However, as an example, sovereign to consumer
+			// changeover chains should utilize customized upgrade handler code similar to below.
+
+			// TODO: should have a way to read from current node home
+			userHomeDir, err := os.UserHomeDir()
+			if err != nil {
+				stdlog.Println("Failed to get home dir %2", err)
+			}
+			nodeHome := userHomeDir + "/.sovereign/config/genesis.json"
+			appState, _, err := genutiltypes.GenesisStateFromGenFile(nodeHome)
+			if err != nil {
+				return fromVM, fmt.Errorf("failed to unmarshal genesis state: %w", err)
+			}
+
+			var consumerGenesis = consumertypes.GenesisState{}
+			appCodec.MustUnmarshalJSON(appState[consumertypes.ModuleName], &consumerGenesis)
+
+			consumerGenesis.PreCCV = true
+			// TODO: confirm InitiGenesis is not automatically called when a module is added during upgrade.
```

Comment:
Could we fix this TODO before merge.

Reply:
Looking into it, there is a duplicate call to InitGenesis. See https://docs.cosmos.network/main/core/upgrade#adding-new-modules-during-upgrades

Reply:
Explanation of why this seems to be a non issue -> https://github.com/cosmos/interchain-security/commit/2a76cd108fcdf414cee64d542bb15b26007e8e19